additionalwiki, additional file magic DB from wiki

unixfreaxjp opened the pull request:
refer to this issue: https://github.com/radare/radare2/issues/10554

radare commented:
My main reason to include new signatures is because they are useful for some cases (forensics mainly) and the newly introduced keys doesn't results in many false positives. didnt tried that, but i would prefer a different filename for this magic file.

unixfreaxjp commented:
> different filename for this magic file.

I really had no idea what to name the new additional db file, I think we should keep it separate to any other file for maybe we will need to trim or merge to some files later on.

Do you have any suggestion for the filename? @radare

unixfreaxjp commented:
Squashed https://github.com/unixfreaxjp/radare2/commit/05d0768ee42d5c8d08fbc7033c625317d43c950e and https://github.com/radare/radare2/pull/11573/commits/d0bea4680d0a8692f1b4f6565e5cb6b2b00315ec into one commit, and pull requesting additional file magic database "extra" to magic signatures in libr/magic/d, ref issue: https://github.com/radare/radare2/issues/10554

unixfreaxjp commented:
This commit compilation were tested without errors in several platform I tests. This commit is only adding additional database for file magic, I son't know the reason why [Travis check stated failed](https://travis-ci.org/radare/radare2/builds/430006772?utm_source=github_status&utm_medium=notification)

In FreeBSD:
```asm
     :
CC rabin2.c
LD rabin2
CC rahash2.c
LD rahash2
CC radiff2.c
LD radiff2
CC radare2.c
LD radare2
CC rafind2.c
LD rafind2
CC rarun2.c
LD rarun2
CC ragg2.c
LD ragg2
CC r2agent.c
LD r2agent
mkdir -p "/usr/share/man/man1"
mkdir -p "/usr/share/man/man7"
for FILE in r2-docker.1 r2agent.1 r2pm.1 rabin2.1 radare2.1 radiff2.1 rafind2.1 ragg2.1 rahash2.1 rarun2.1 rasm2.1 rax2.1 ; do \
 ln -fs "/usr/home/radare2dev/man/$FILE" "/usr/share/man/man1/$FILE" ; done
cd "/usr/share/man/man1" && ln -fs radare2.1 r2.1
for FILE in *.7 ; do \
 ln -fs "/usr/home/radare2dev/man/$FILE" "/usr/share/man/man7/$FILE" ; done
/usr/bin/install -d "/usr/share/doc/radare2"
for FILE in 3D all68k.r2 android.md avr.md bash_autocompletion.sh brainfuck.md calling-conventions.md capstone.md clang-format crosscompile debug-internals.md debug.md dexdump.sh dospart dwarf-line.rsc emscripten.md esil.md flirt.md fortunes.creepy fortunes.fun fortunes.nsfw fortunes.tips freebsd.md gdb.md gprobe.md haiku.md hud ida.md images intro.md io-fixing.md io.md iocache ios.md iphone.md kvast.md lang-plugins license.md macos.md macros maemo.md node.js objc.pl oo.md openbsd.md pdb plugins.md portscan.r2 qnx.md r2ascii.txt rap.md rapatch.md releases.md repo rgraph.md sandbox sdb_ordinal.md siol.md solaris.md static-plugins statis.md strings.md termux.md uncrustify.cfg vim vim2r2.js windbg.md windows.md yara.md zsh ; do \
 ln -fs "/usr/home/radare2dev/doc/$FILE" "/usr/share/doc/radare2" ; done
for FILE in r_anal.pc r_asm.pc r_bin.pc r_bp.pc r_config.pc r_cons.pc r_core.pc r_crypto.pc r_debug.pc r_egg.pc r_flag.pc r_fs.pc r_hash.pc r_io.pc r_lang.pc r_magic.pc r_parse.pc r_reg.pc r_search.pc r_socket.pc r_syscall.pc r_util.pc ; do \
 ln -fs "/usr/home/radare2dev/pkgcfg/$FILE" "/usr/lib/pkgconfig/$FILE" ; done
rm -rf "/usr/share/radare2/3.0.0-git/www"
rm -rf "/usr/lib/radare2/3.0.0-git/www" # old dir
mkdir -p "/usr/share/radare2/3.0.0-git/www"
for FILE in console.html d3 enyo f favicon.ico graph index.html index.old.html log.html m old p r2.svg rlogo.png t upload.html ; do \
 ln -fs "/usr/home/radare2dev/shlr/www/$FILE" "/usr/share/radare2/3.0.0-git/www" ; done
for DIR in libr/cons/d libr/flag/d libr/bin/d libr/asm/d libr/syscall/d libr/magic/d libr/anal/d ; do (\
 cd "$DIR" ; \
 echo "$DIR" ; \
 gmake install-symlink ); \
done
libr/cons/d
gmake[1]: Entering directory '/usr/home/radare2dev/libr/cons/d'
mkdir -p "/usr/share/radare2/3.0.0-git/cons"
for FILE in * ; do \
 if [ $FILE != Makefile -a $FILE != meson.build -a -f $FILE ]; then \
  ln -fs "/usr/home/radare2dev/libr/cons/d/$FILE" "/usr/share/radare2/3.0.0-git/cons/$FILE" ; \
 fi ; \
done
gmake[1]: Leaving directory '/usr/home/radare2dev/libr/cons/d'
libr/flag/d
gmake[1]: Entering directory '/usr/home/radare2dev/libr/flag/d'
rm -f *.r2
for a in alloc env fs network process stdout string ; do b=$(echo `cat $a`); echo "ft $a $b" ; done > tags.r2
mkdir -p "/usr/share/radare2/3.0.0-git/flag"
ln -fs /usr/home/radare2dev/libr/flag/d/tags.r2 /usr/share/radare2/3.0.0-git/flag/tags.r2
gmake[1]: Leaving directory '/usr/home/radare2dev/libr/flag/d'
libr/bin/d
gmake[1]: Entering directory '/usr/home/radare2dev/libr/bin/d'
mkdir -p "/usr/share/radare2/3.0.0-git/format"
for FILE in * ; do \
 if [ "$FILE" != Makefile ]; then \
  ln -fs "/usr/home/radare2dev/libr/bin/d/$FILE" "/usr/share/radare2/3.0.0-git/format/$FILE" ; \
 fi ; \
done
gmake[1]: Leaving directory '/usr/home/radare2dev/libr/bin/d'
libr/asm/d
gmake[1]: Entering directory '/usr/home/radare2dev/libr/asm/d'
mkdir -p "/usr/share/radare2/3.0.0-git/opcodes"
for FILE in *.sdb ; do ln -fs "/usr/home/radare2dev/libr/asm/d/$FILE" "/usr/share/radare2/3.0.0-git/opcodes/$FILE" ; done
gmake[1]: Leaving directory '/usr/home/radare2dev/libr/asm/d'
libr/syscall/d
gmake[1]: Entering directory '/usr/home/radare2dev/libr/syscall/d'
mkdir -p "/usr/share/radare2/3.0.0-git/syscall"
for FILE in *.sdb ; do ln -fs "/usr/home/radare2dev/libr/syscall/d/$FILE" "/usr/share/radare2/3.0.0-git/syscall/$FILE" ; done
gmake[1]: Leaving directory '/usr/home/radare2dev/libr/syscall/d'
libr/magic/d
gmake[1]: Entering directory '/usr/home/radare2dev/libr/magic/d'
mkdir -p "/usr/share/radare2/3.0.0-git/magic"
for FILE in OpenBSD archive cafebabe code compressed database elf extras filesystems firmware flash freebsd gimp gpg html images java jpeg linux mach macintosh mail.news microsoft mime netbsd network pdf perl python riff sniffer sql ssh uuencode video vim vorbis zfs ; do \
 ln -fs "/usr/home/radare2dev/libr/magic/d/default/$FILE" "/usr/share/radare2/3.0.0-git/magic/$FILE" ; \
done
gmake[1]: Leaving directory '/usr/home/radare2dev/libr/magic/d'
libr/anal/d
gmake[1]: Entering directory '/usr/home/radare2dev/libr/anal/d'
cd ../../.. && mkdir -p "/usr/share/radare2/3.0.0-git/fcnsign" && \
for FILE in cc-arm-32.sdb cc-arm-64.sdb cc-avr-8.sdb cc-m68k-32.sdb cc-mips-32.sdb cc-powerpc-32.sdb cc-powerpc-64.sdb cc-riscv-64.sdb cc-sparc-32.sdb cc-x86-32.sdb cc-x86-64.sdb cc-xtensa-32.sdb spec.sdb types-16.sdb types-32.sdb types-64.sdb types-arm-ios-16.sdb types-arm-ios-32.sdb types-arm-ios-64.sdb types-linux.sdb types-windows.sdb types-x86-macos-64.sdb types-x86-windows-32.sdb types-x86-windows-64.sdb types.sdb; do \
 ln -fs "/usr/home/radare2dev/libr/anal/d/$FILE" "/usr/share/radare2/3.0.0-git/fcnsign/$FILE" ; \
done
gmake[1]: Leaving directory '/usr/home/radare2dev/libr/anal/d'
cd libr && gmake install-symlink
gmake[1]: Entering directory '/usr/home/radare2dev/libr'
gmake[2]: Entering directory '/usr/home/radare2dev'
for FILE in r_anal.pc r_asm.pc r_bin.pc r_bp.pc r_config.pc r_cons.pc r_core.pc r_crypto.pc r_debug.pc r_egg.pc r_flag.pc r_fs.pc r_hash.pc r_io.pc r_lang.pc r_magic.pc r_parse.pc r_reg.pc r_search.pc r_socket.pc r_syscall.pc r_util.pc ; do \
 ln -fs "/usr/home/radare2dev/pkgcfg/$FILE" "/usr/lib/pkgconfig/$FILE" ; done
gmake[2]: Leaving directory '/usr/home/radare2dev'
gmake[1]: Leaving directory '/usr/home/radare2dev/libr'
cd binr && gmake install-symlink
gmake[1]: Entering directory '/usr/home/radare2dev/binr'
cd .. && \
mkdir -p "/usr/bin" && \
for BINARY in rax2 rasm2 rabin2 rahash2 radiff2 radare2 rafind2 rarun2 ragg2 r2agent ; do ln -fs "/usr/home/radare2dev/binr/$BINARY/$BINARY" "/usr/bin/$BINARY" ; done
cd .. && ln -fs "/usr/home/radare2dev/binr/r2pm/r2pm" "/usr/bin/r2pm"
cd .. && rm -rf "/usr/share/radare2/3.0.0-git/r2pm"
cd .. && mkdir -p "/usr/share/radare2/3.0.0-git/"
#ln -fs "/usr/home/radare2dev/binr/r2pm/d" "/usr/share/radare2/3.0.0-git/r2pm"
cd .. && ln -fs "/usr/bin/radare2" "/usr/bin/r2"
cd .. && ln -fs "/usr/home/radare2dev/binr/preload/libr2.so" "/usr/lib/libr2.so"
cd .. && cd "/usr/lib" && \
 mv libr2.so libr2.so.3.0.0-git && \
 ln -fs libr2.so.3.0.0-git libr2.so
gmake[1]: Leaving directory '/usr/home/radare2dev/binr'
cd shlr && gmake install-symlink
gmake[1]: Entering directory '/usr/home/radare2dev/shlr'
mkdir -p "/usr/lib"
#cd tcc ; for FILE in libr_tcc* ; do \
# ln -fs `pwd`/$FILE /usr/lib/$FILE ; done
gmake[1]: Leaving directory '/usr/home/radare2dev/shlr'
mkdir -p "/usr/bin"
ln -fs "/usr/home/radare2dev/sys/indent.sh" "/usr/bin/r2-indent"
ln -fs "/usr/home/radare2dev/sys/r2-docker.sh" "/usr/bin/r2-docker"
mkdir -p "/usr/share/radare2/3.0.0-git/hud"
ln -fs "/usr/home/radare2dev/doc/hud" "/usr/share/radare2/3.0.0-git/hud/main"
#mkdir -p "/usr/share/radare2/3.0.0-git/flag"
#ln -fs /usr/home/radare2dev/libr/flag/d/tags.r2 "/usr/share/radare2/3.0.0-git/flag/tags.r2"
cd "/usr/lib/radare2/" ;\
 rm -f last ; ln -fs 3.0.0-git last
cd "/usr/share/radare2/" ;\
 rm -f last ; ln -fs 3.0.0-git last
mkdir -p "/usr/share/radare2/3.0.0-git/"
/bin/sh sys/ldconfig.sh
/bin/sh ./configure-plugins --rm-static //usr/lib/radare2/last/
configure-plugins: Loading ./plugins.cfg ..
Removed 0 shared plugins that are already static
$
$
$ r2 -v
radare2 3.0.0-git 19485 @ freebsd-x86-32 git.2.9.0-146-gd0bea46
commit: d0bea4680d0a8692f1b4f6565e5cb6b2b00315ec build: 2018-09-19__07:11:03
```
and Linux:
```asm
  :
CC rasm2.c
LD rasm2
CC rabin2.c
LD rabin2
CC rahash2.c
LD rahash2
CC radiff2.c
LD radiff2
CC radare2.c
radare2.c: In function 'main':
radare2.c:475:6: warning: variable 'prefile' set but not used [-Wunused-but-set-variable]
  int prefile = 0;
      ^
LD radare2
CC rafind2.c
LD rafind2
CC rarun2.c
LD rarun2
CC ragg2.c
LD ragg2
CC r2agent.c
LD r2agent
mkdir -p "/usr/share/man/man1"
mkdir -p "/usr/share/man/man7"
for FILE in r2-docker.1 r2agent.1 r2pm.1 rabin2.1 radare2.1 radiff2.1 rafind2.1 ragg2.1 rahash2.1 rarun2.1 rasm2.1 rax2.1 ; do \
 ln -fs "/home/radare2dev/man/$FILE" "/usr/share/man/man1/$FILE" ; done
cd "/usr/share/man/man1" && ln -fs radare2.1 r2.1
for FILE in *.7 ; do \
 ln -fs "/home/radare2dev/man/$FILE" "/usr/share/man/man7/$FILE" ; done
/usr/bin/install -d "/usr/share/doc/radare2"
for FILE in 3D all68k.r2 android.md avr.md bash_autocompletion.sh brainfuck.md calling-conventions.md capstone.md clang-format crosscompile debug-internals.md debug.md dexdump.sh dospart dwarf-line.rsc emscripten.md esil.md flirt.md fortunes.creepy fortunes.fun fortunes.nsfw fortunes.tips freebsd.md gdb.md gprobe.md haiku.md hud ida.md images intro.md io-fixing.md io.md iocache ios.md iphone.md kvast.md lang-plugins license.md macos.md macros maemo.md node.js objc.pl oo.md openbsd.md pdb plugins.md portscan.r2 qnx.md r2ascii.txt rap.md rapatch.md releases.md repo rgraph.md sandbox sdb_ordinal.md siol.md solaris.md static-plugins statis.md strings.md termux.md uncrustify.cfg vim vim2r2.js windbg.md windows.md yara.md zsh ; do \
 ln -fs "/home/radare2dev/doc/$FILE" "/usr/share/doc/radare2" ; done
for FILE in r_anal.pc r_asm.pc r_bin.pc r_bp.pc r_config.pc r_cons.pc r_core.pc r_crypto.pc r_debug.pc r_egg.pc r_flag.pc r_fs.pc r_hash.pc r_io.pc r_lang.pc r_magic.pc r_parse.pc r_reg.pc r_search.pc r_socket.pc r_syscall.pc r_util.pc ; do \
 ln -fs "/home/radare2dev/pkgcfg/$FILE" "/usr/lib/pkgconfig/$FILE" ; done
rm -rf "/usr/share/radare2/3.0.0-git/www"
rm -rf "/usr/lib/radare2/3.0.0-git/www" # old dir
mkdir -p "/usr/share/radare2/3.0.0-git/www"
for FILE in console.html d3 enyo f favicon.ico graph index.html index.old.html log.html m old p r2.svg rlogo.png t upload.html ; do \
 ln -fs "/home/radare2dev/shlr/www/$FILE" "/usr/share/radare2/3.0.0-git/www" ; done
for DIR in libr/cons/d libr/flag/d libr/bin/d libr/asm/d libr/syscall/d libr/magic/d libr/anal/d ; do (\
 cd "$DIR" ; \
 echo "$DIR" ; \
 make install-symlink ); \
done
libr/cons/d
make[1]: Entering directory '/home/radare2dev/libr/cons/d'
mkdir -p "/usr/share/radare2/3.0.0-git/cons"
for FILE in * ; do \
 if [ $FILE != Makefile -a $FILE != meson.build -a -f $FILE ]; then \
     ln -fs "/home/radare2dev/libr/cons/d/$FILE" "/usr/share/radare2/3.0.0-git/cons/$FILE" ; \
 fi ; \
done
make[1]: Leaving directory '/home/radare2dev/libr/cons/d'
libr/flag/d
make[1]: Entering directory '/home/radare2dev/libr/flag/d'
rm -f *.r2
for a in alloc env fs network process stdout string ; do b=$(echo `cat $a`); echo "ft $a $b" ; done > tags.r2
mkdir -p "/usr/share/radare2/3.0.0-git/flag"
ln -fs /home/radare2dev/libr/flag/d/tags.r2 /usr/share/radare2/3.0.0-git/flag/tags.r2
make[1]: Leaving directory '/home/radare2dev/libr/flag/d'
libr/bin/d
make[1]: Entering directory '/home/radare2dev/libr/bin/d'
mkdir -p "/usr/share/radare2/3.0.0-git/format"
for FILE in * ; do \
 if [ "$FILE" != Makefile ]; then \
     ln -fs "/home/radare2dev/libr/bin/d/$FILE" "/usr/share/radare2/3.0.0-git/format/$FILE" ; \
 fi ; \
done
make[1]: Leaving directory '/home/radare2dev/libr/bin/d'
libr/asm/d
make[1]: Entering directory '/home/radare2dev/libr/asm/d'
mkdir -p "/usr/share/radare2/3.0.0-git/opcodes"
for FILE in *.sdb ; do ln -fs "/home/radare2dev/libr/asm/d/$FILE" "/usr/share/radare2/3.0.0-git/opcodes/$FILE" ; done
make[1]: Leaving directory '/home/radare2dev/libr/asm/d'
libr/syscall/d
make[1]: Entering directory '/home/radare2dev/libr/syscall/d'
mkdir -p "/usr/share/radare2/3.0.0-git/syscall"
for FILE in *.sdb ; do ln -fs "/home/radare2dev/libr/syscall/d/$FILE" "/usr/share/radare2/3.0.0-git/syscall/$FILE" ; done
make[1]: Leaving directory '/home/radare2dev/libr/syscall/d'
libr/magic/d
make[1]: Entering directory '/home/radare2dev/libr/magic/d'
mkdir -p "/usr/share/radare2/3.0.0-git/magic"
for FILE in OpenBSD archive cafebabe code compressed database elf extras filesystems firmware flash freebsd gimp gpg html images java jpeg linux mach macintosh mail.news microsoft mime netbsd network pdf perl python riff sniffer sql ssh uuencode video vim vorbis zfs ; do \
 ln -fs "/home/radare2dev/libr/magic/d/default/$FILE" "/usr/share/radare2/3.0.0-git/magic/$FILE" ; \
done
make[1]: Leaving directory '/home/radare2dev/libr/magic/d'
libr/anal/d
make[1]: Entering directory '/home/radare2dev/libr/anal/d'
cd ../../.. && mkdir -p "/usr/share/radare2/3.0.0-git/fcnsign" && \
for FILE in cc-arm-32.sdb cc-arm-64.sdb cc-avr-8.sdb cc-m68k-32.sdb cc-mips-32.sdb cc-powerpc-32.sdb cc-powerpc-64.sdb cc-riscv-64.sdb cc-sparc-32.sdb cc-x86-32.sdb cc-x86-64.sdb cc-xtensa-32.sdb spec.sdb types-16.sdb types-32.sdb types-64.sdb types-arm-ios-16.sdb types-arm-ios-32.sdb types-arm-ios-64.sdb types-linux.sdb types-windows.sdb types-x86-macos-64.sdb types-x86-windows-32.sdb types-x86-windows-64.sdb types.sdb; do \
 ln -fs "/home/radare2dev/libr/anal/d/$FILE" "/usr/share/radare2/3.0.0-git/fcnsign/$FILE" ; \
done
make[1]: Leaving directory '/home/radare2dev/libr/anal/d'
cd libr && make install-symlink
make[1]: Entering directory '/home/radare2dev/libr'
make[2]: Entering directory '/home/rik/radare2'
for FILE in r_anal.pc r_asm.pc r_bin.pc r_bp.pc r_config.pc r_cons.pc r_core.pc r_crypto.pc r_debug.pc r_egg.pc r_flag.pc r_fs.pc r_hash.pc r_io.pc r_lang.pc r_magic.pc r_parse.pc r_reg.pc r_search.pc r_socket.pc r_syscall.pc r_util.pc ; do \
 ln -fs "/home/radare2dev/pkgcfg/$FILE" "/usr/lib/pkgconfig/$FILE" ; done
make[2]: Leaving directory '/home/rik/radare2'
make[1]: Leaving directory '/home/radare2dev/libr'
cd binr && make install-symlink
make[1]: Entering directory '/home/radare2dev/binr'
cd .. && \
mkdir -p "/usr/bin" && \
for BINARY in rax2 rasm2 rabin2 rahash2 radiff2 radare2 rafind2 rarun2 ragg2 r2agent ; do ln -fs "/home/radare2dev/binr/$BINARY/$BINARY" "/usr/bin/$BINARY" ; done
cd .. && ln -fs "/home/radare2dev/binr/r2pm/r2pm" "/usr/bin/r2pm"
cd .. && rm -rf "/usr/share/radare2/3.0.0-git/r2pm"
cd .. && mkdir -p "/usr/share/radare2/3.0.0-git/"
#ln -fs "/home/radare2dev/binr/r2pm/d" "/usr/share/radare2/3.0.0-git/r2pm"
cd .. && ln -fs "/usr/bin/radare2" "/usr/bin/r2"
cd .. && ln -fs "/home/radare2dev/binr/preload/libr2.so" "/usr/lib/libr2.so"
cd .. && cd "/usr/lib" && \
 mv libr2.so libr2.so.3.0.0-git && \
 ln -fs libr2.so.3.0.0-git libr2.so
make[1]: Leaving directory '/home/radare2dev/binr'
cd shlr && make install-symlink
make[1]: Entering directory '/home/radare2dev/shlr'
mkdir -p "/usr/lib"
#cd tcc ; for FILE in libr_tcc* ; do \
#    ln -fs `pwd`/$FILE /usr/lib/$FILE ; done
make[1]: Leaving directory '/home/radare2dev/shlr'
mkdir -p "/usr/bin"
ln -fs "/home/radare2dev/sys/indent.sh" "/usr/bin/r2-indent"
ln -fs "/home/radare2dev/sys/r2-docker.sh" "/usr/bin/r2-docker"
mkdir -p "/usr/share/radare2/3.0.0-git/hud"
ln -fs "/home/radare2dev/doc/hud" "/usr/share/radare2/3.0.0-git/hud/main"
#mkdir -p "/usr/share/radare2/3.0.0-git/flag"
#ln -fs /home/radare2dev/libr/flag/d/tags.r2 "/usr/share/radare2/3.0.0-git/flag/tags.r2"
cd "/usr/lib/radare2/" ;\
 rm -f last ; ln -fs 3.0.0-git last
cd "/usr/share/radare2/" ;\
 rm -f last ; ln -fs 3.0.0-git last
mkdir -p "/usr/share/radare2/3.0.0-git/"
/bin/sh sys/ldconfig.sh
/bin/sh ./configure-plugins --rm-static //usr/lib/radare2/last/
configure-plugins: Loading ./plugins.cfg ..
Removed 0 shared plugins that are already static
$
$
$ r2 -v
radare2 3.0.0-git 19485 @ linux-x86-64 git.2.9.0-146-gd0bea46
commit: d0bea4680d0a8692f1b4f6565e5cb6b2b00315ec build: 2018-09-19__07:24:17
```

Maijin commented:
Greetings,

To determine what caused the RED state, you can search the Travis log by searching "[XX]" in the log with cmd+f.

According to this log it seems that your pull-requests was adding additional detection for the same magic as OSX mach-o tested in the 2 tests below:

<img width="882" alt="screen shot 2018-09-19 at 08 29 52" src="https://user-images.githubusercontent.com/1408600/45734770-fa893100-bbe5-11e8-911a-58bebb043f6d.png">

Given the rule "Palm Desktop Calendar Archive" seems to produce false positive (it's redundant with  OSX mach-o detection), you can probably remove it from the list you created and it should be green ✅.

Regards,

radare commented:
The other magic signatures seem not interesting in 2018 for forensics or too specific and not much defined. That will make most users get false results of things they will probably not be interested in

unixfreaxjp commented:
> That will make most users get false results of things they will probably not be interested

Yes, agreed.
The Wikipedia reference was not containing too much information on each magic too. 
I don't think it doesn't bring too much advantages for us to have these new signature.

>  will make most users get false results

to be very honest. Many false results came up during test, especially if we tried to /m for memory dump or firmware big data. I omit many of the sig itself. Furthermore, he interesting sigs are all already included in the previous magic database. 

I don't know if we must continue to add this or not, it doesn't hurt much to not including these. Yet the r2 users can add that "extras" dataset if they really need it as "optional"

unixfreaxjp commented:
> You can update your Pull-request by changing the commit and then do a push -f in local fork, it will update here and relaunch the test automatically, you do not need to close the PR and reopen a new one.
> 
> Regards,

Hello @Maijin thank you for the analysis and advise, I am trying to sort out your advise and @radare advise altogether now, and by seeing the leftover.. it doesn't look so much interesting to add this "extras" to be very honest. Let me know if we want to have a GO for this as additional feature.

Regards

unixfreaxjp commented on libr/magic/d/default/extras:
```diff
@@ -0,0 +1,79 @@
+0	lelong	0x425047FB	Better Portable Graphics 
+0	belong	0x762F3101	OpenEXR image
+0	lelong	0x762F3101	OpenEXR image
+0	belong	0x25215053	PostScript document file 
```
For the Postscript it is difficult, since more than 4bytes are random.. so far I tests.
It is hard to find OpenEXR image format file so I download several samples from internet to check only.

> lelong	0x425047FB	Better Portable Graphics 

This causing a bit of false positive. Not as many as others that were being omitted, but still false positive exists.

unixfreaxjp commented on libr/magic/d/default/extras:
```diff
@@ -0,0 +1,79 @@
+0	belong	0xedabeedb	RedHat RPM package
+0	lelong	0xedabeedb	RedHat RPM package
+0	belong	0x53503031	Amazon Kindle Update Package
+0	lelong	0x53503031	Amazon Kindle Update Package
```
RPM is good I guess, but I am not sure whether we need to forensics scanning RPM.
Kindly update package... I have no idea  not thinking this is a must, but it is in the wiki and not having false positive so I included. I don't mind to exclude these list.

unixfreaxjp commented on libr/magic/d/default/extras:
```diff
@@ -0,0 +1,79 @@
+0	belong	0xbebafeca	Palm Desktop Calendar Archive 
+0	lelong	0xbebafeca	Palm Desktop Calendar Archive 
+0	belong	0x00014244	Palm Desktop To Do Archive 
+0	lelong	0x00014244	Palm Desktop To Do Archive 
```
> All those palm* may be removed palm is dead

No problem, was thinking the same on the Palm is dead,.

> because will spot lot of false positives and 

I was suspecting a lot of false positive actually, but it was not during the check, maybe the sample range for tests need to be improved wider too. I don't mind to remove these.

unixfreaxjp commented on libr/magic/d/default/extras:
```diff
@@ -0,0 +1,79 @@
+# blaim - unixfreaxjp for incorrect approach -Tue Sep 18 05:58:29 JST 2018
+#
+
+0	string	\x42\x41\x43\x4B\x4D\x49\x4B\x45\x44\x49\x53\x4B	AmiBack backup file
```
> AmiBack backup file

well, I was smiling  when seeing this magic.. Seriously no one will run r2 for checking tape or floppy (that big one) backup file... And AmiBack was  a lot in them..

unixfreaxjp commented:
@radare 

> From this list i would only take:
> 
> - lua bytecode (which is dupped, and probably need more than 4 bytes check)
> - google chrome extension (probably will have lot of false positives))
> - postscript
l> - ibpcap/pcapng

Optionally, (1) for this small addition maybe we shouldn't add as new file, but to add to previous existing files, it's too small for "extras", or, maybe it is an option (2) to pending this addition until we got more data to add. Please direct us to proceed. Thank you

unixfreaxjp commented:
No replies. Waited. I consider it as "No go" then.

XVilka commented:
@unixfreaxjp @Maijin @radare I would consider a different approach - pack them into r2pm as an additional package, probably also with magics from https://github.com/ReFirmLabs/binwalk

unixfreaxjp commented:
@XVilka @radare @Maijin 

That wiki is having too few data. I think I agreed with @XVilka to drop this, and renew this approach, to really "improve" r2 magic with more additional solid magic.  It will be a serious load of work though.. I was thinking I could add that wiki one for some couple hours work due to can't take much time away from day work handling.

Let me rant a bit. I sometimes wonder why we still need "file magic basis signature" since not many people know how to make/maintain it... and BTW we have 2 other better option: Zignature or Yara..  It sounds a bit nice if /m will use  Yara db or zignature db instead maybe. Many people know these formats, it is good for maintaining them, No? Bad idea? 

This is a discussion yes? Seriously,  if we want to have a real forensics signature maybe big database will come eventually, like XWays for instance, they have about 40,000 data. Encase also have big DB too. ps: both are not open source. Someday we will need to face bigger amount than now too. The more well-known sig format maybe a good idea.

>  I would consider a different approach - pack them into r2pm as an additional package

This looks very interesting.

XVilka commented:
R2 has Yara support, just might be enhanced: `r2pm -i r2yara` will install
plugin which will add `yara` commands into your radare2.

On Fri, Sep 21, 2018, 2:37 AM unixfreaxjp <notifications@github.com> wrote:

> @XVilka <https://github.com/XVilka> @radare <https://github.com/radare>
> @Maijin <https://github.com/Maijin>
>
> That wiki is having too few data. I think I agreed with @XVilka
> <https://github.com/XVilka> to drop this, and renew this approach, to
> really "improve" r2 magic with more additional solid magic. It will be a
> serious load of work though.. I was thinking I could add that wiki one for
> some couple hours work due to can't take much time away from day work
> handling.
>
> Let me rant a bit. I sometimes wonder why we still need "file magic basis
> signature" since not many people know how to make/maintain it... and BTW we
> have 2 other better option: Zignature or Yara.. It sounds a bit nice if /m
> will use Yara db or zignature db instead maybe. Many people know these
> formats, it is good for maintaining them, No? Bad idea?
>
> This is a discussion yes? Seriously, if we want to have a real forensics
> signature maybe big database will come eventually, like XWays for instance,
> they have about 40,000 data. Encase also have big DB too. ps: both are not
> open source. Someday we will need to face bigger amount than now too. The
> more well-known sig format maybe a good idea.
>
> I would consider a different approach - pack them into r2pm as an
> additional package
>
> This looks very interesting.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/radare/radare2/pull/11573#issuecomment-423288976>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAMZ_aZyI3TkFXhxO6kxciLHff5vV_9Cks5uc-BZgaJpZM4WtPmD>
> .
>

unixfreaxjp commented:
> 
> R2 has Yara support, just might be enhanced: `r2pm -i r2yara` will install plugin which will add `yara` commands into your radare2.
I know that, is the carve function that I mean (below)

> I sometimes wonder why we still need "file magic basis signature" since nowadays not many people know how to make/maintain it... and BTW r2 have 2 other better sig option too: Zignatureor Yara.. It sounds nice if /m will use Yaradb or Zignaturedb instead maybe? Many people know these formats, it is good resource for maintaining them. No? Bad idea?

radare commented:
/m is shipped with r2 and its a unix standard. Yara is an external dependency, its slow and bloated compared to how magic signatures work. Also, magic search allows recursive searching. Not knowing something about r2 is not a reason for removing it. I use it in many situations where yara wouldnt be possible to use. For example when using it on embedded devices or when searching in memory. Yara requires files to work with

The purpose of zignatures is completely different than magic headers. So i dont see how they may overlap in functionality.

> On 22 Sep 2018, at 10:49, unixfreaxjp <notifications@github.com> wrote:
> 
> R2 has Yara support, just might be enhanced: r2pm -i r2yara will install plugin which will add yara commands into your radare2.
> I know that, is the carve function that I mean (below)
> 
> I sometimes wonder why we still need "file magic basis signature" since nowadays not many people know how to make/maintain it... and BTW r2 have 2 other better sig option too: Zignatureor Yara.. It sounds nice if /m will use Yaradb or Zignaturedb instead maybe? Many people know these formats, it is good resource for maintaining them. No? Bad idea?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

radare commented:
You can think of /m as a binwalk replacement with support for multiple io backends

> On 22 Sep 2018, at 10:49, unixfreaxjp <notifications@github.com> wrote:
> 
> R2 has Yara support, just might be enhanced: r2pm -i r2yara will install plugin which will add yara commands into your radare2.
> I know that, is the carve function that I mean (below)
> 
> I sometimes wonder why we still need "file magic basis signature" since nowadays not many people know how to make/maintain it... and BTW r2 have 2 other better sig option too: Zignatureor Yara.. It sounds nice if /m will use Yaradb or Zignaturedb instead maybe? Many people know these formats, it is good resource for maintaining them. No? Bad idea?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

XVilka commented:
Yara can scan memory

On Sat, Sep 22, 2018, 5:00 PM radare <notifications@github.com> wrote:

> /m is shipped with r2 and its a unix standard. Yara is an external
> dependency, its slow and bloated compared to how magic signatures work.
> Also, magic search allows recursive searching. Not knowing something about
> r2 is not a reason for removing it. I use it in many situations where yara
> wouldnt be possible to use. For example when using it on embedded devices
> or when searching in memory. Yara requires files to work with
>
> The purpose of zignatures is completely different than magic headers. So i
> dont see how they may overlap in functionality.
>
> > On 22 Sep 2018, at 10:49, unixfreaxjp <notifications@github.com> wrote:
> >
> > R2 has Yara support, just might be enhanced: r2pm -i r2yara will install
> plugin which will add yara commands into your radare2.
> > I know that, is the carve function that I mean (below)
> >
> > I sometimes wonder why we still need "file magic basis signature" since
> nowadays not many people know how to make/maintain it... and BTW r2 have 2
> other better sig option too: Zignatureor Yara.. It sounds nice if /m will
> use Yaradb or Zignaturedb instead maybe? Many people know these formats, it
> is good resource for maintaining them. No? Bad idea?
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub, or mute the thread.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/radare/radare2/pull/11573#issuecomment-423728909>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAMZ_U16TbwYzNEFQExGCg61Z5a2Kfo-ks5udfw3gaJpZM4WtPmD>
> .
>

unixfreaxjp commented:
> Yara can scan memory

I must admit, that loading all necessary Yara sigs to scan all possible magics in, say, embedded environment is like making an OOM  chaos.. for /m it is a snap operation.
